jgrapht-core/src/test/java/org/jgrapht/graph/SerializationTest.java

```diff
@@ -17,77 +17,660 @@
+ * </ul>
  *
  * @author John V. Sichi
  */
 public class SerializationTest
 {
-    // ~ Instance fields --------------------------------------------------------
+    private static final String v1 = "v1";
+    private static final String v2 = "v2";
+    private static final String v3 = "v3";
+    private static final List<String> vertexList = Arrays.asList(v1, v2, v3);
+    private static final List<List<String>> vertexPairs = Arrays
+        .asList(Arrays.asList(v1, v2), Arrays.asList(v2, v1), Arrays.asList(v1, v3),
+            Arrays.asList(v3, v1), Arrays.asList(v2, v3), Arrays.asList(v3, v2));
+
+    public static <V, E> void assertContainsAllVertices(Graph<V, E> graph, List<V> vertices)
+    {
+        for (V v : vertices) {
+            assertTrue(graph.containsVertex(v));
+        }
+    }
+
+    public static <V, E> void checkEdgesOf(Graph<V, E> graph, List<Integer> edges, List<V> vertices)
+    {
+        if (edges.size() != vertices.size()) {
+            throw new IllegalArgumentException(
+                "the size of list of #edges and vertices should match");
+        }
+        for (int i = 0; i < edges.size(); i++) {
+            assertEquals(edges.get(i).intValue(), graph.edgesOf(vertices.get(i)).size());
+        }
+    }
+
+    public static <E> void assertAllEdges(Graph<String, E> graph1, Graph<String, E> graph2)
+    {
+        for (int i = 0; i < vertexPairs.size(); i++) {
+            String a = vertexPairs.get(i).get(0);
+            String b = vertexPairs.get(i).get(1);
+            assertEquals(graph1.getAllEdges(a, b).size(), graph2.getAllEdges(a, b).size());
+            assertEquals(graph1.containsEdge(a, b), graph2.containsEdge(a, b));
+        }
+    }
+
+    private static <E> void verifyBasic(
+        Graph<String, E> graph1, Graph<String, E> graph2, List<Integer> numberOfEdges)
+    {
+        assertContainsAllVertices(graph2, vertexList);
+        assertContainsAllVertices(graph1, vertexList);
+
+        assertAllEdges(graph1, graph2);
+
+        checkEdgesOf(graph2, numberOfEdges, vertexList);
+        checkEdgesOf(graph1, numberOfEdges, vertexList);
+
+        assertEquals(graph1.toString(), graph2.toString());
+    }
+
+    private static <E> void assertWeight(
+        Graph<String, E> graph, List<Double> weights, String vertex1, String vertex2)
+    {
+        Set<E> edgeSet = graph.getAllEdges(vertex1, vertex2);
+        for (E e : edgeSet)
+            assertTrue(e instanceof DefaultWeightedEdge);
+        assertEquals(new HashSet<>(weights),
+            edgeSet.stream().map(e -> (DefaultWeightedEdge) e).map(DefaultWeightedEdge::getWeight)
+                .collect(Collectors.toSet()));
+    }
+
+    /**
+     * Tests serialization of {@link SimpleGraph}.
+     * <p>
+     * undirected
+     * no self-loop
+     * no multiple-edges
+     * unweighted
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testSerialization_SimpleGraph()
```

Comment:
Since the name of this class is already SerializationTest, you don't need to include Serialization in the method name...just testSimpleGraph would be good enough.  (In our coding conventions, we don't use underscores in method names.)  Same for all the other test methods.

Reply:
Thanks for the suggestion. I've changed all the methods as you suggested.